templates/search_form.tt.html

```diff
@@ -0,0 +1,233 @@
+<h1> [% lang('search_products') %] </h1>
+<form method="post" action="/cgi/search.pl" enctype="multipart/form-data" id="search_form">
+
+    <!-- Word Search -->
+    <div class="row">
+        <div class="large-12 columns">
+            <label for="search_terms2">[% lang ('search_terms_note') %]</label>
+            <input type="text" name="search_terms2" id="search_terms2" value="[%search_terms%]" />
+	    </div>
+    </div>
+
+    <!-- Criteria -->
+
+    <h3>[% lang('search_tags') %]</h3>
+    <label>[% lang('search_criteria') %]</label>
+
+    <!-- Criteria form -->
+
+    <div class="row">
+        [% FOREACH type IN criteria %]
+            <div class="small-12 medium-12 large-6 columns criterion-row" style="padding-top:1rem">
+                <div class="row">
+                    
+                    <div class="small-12 medium-12 large-5 columns">
+                                                
+                        <select name="tagtype_${type.id}" id="tagtype_${type.id}">
+                            [% FOREACH value IN type.type_arrays %]
+                                [% IF value.type_labels == 'choose a criterion...' %]
+                                    <option value="[% value.type_values %]" label="[% value.type_labels %]" selected="selected"></option>
```

Comment:
The selected value is not necessarily the first one. You probably need to compare value.type_value to the type_value variable you are also passing:

```
		push @{$template_data_ref->{criteria}}, {
			id => $i,
			type_value => $search_tags[$i][0],
			type_arrays => \@type_array,
			contain_value => $search_tags[$i][1],
			contain_array => \@contains,
			input_value => $search_tags[$i][2],
		};

Maybe to make it clearer:

- type_value could be renamed to selected_type_value
- type_arrays could be renamed to type_options, with inside the array of { value => .. , label => .. }


Same for the other parts of the form below.

Reply:
Alright. I'll make these changes